tgui/packages/tgui/interfaces/Uplink.js

```diff
@@ -79,15 +87,35 @@ export const Uplink = (props, context) => {
 
 const ItemsPage = (_properties, context) => {
   const { act, data } = useBackend(context);
-  const {
-    crystals,
-    cats,
-  } = data;
+  const { crystals, cats } = data;
   // Default to first
-  const [
-    uplinkCat,
-    setUplinkCat,
-  ] = useLocalState(context, 'uplinkTab', cats[0]);
+  const [uplinkCat, setUplinkCat] = useLocalState(
+    context,
+    "uplinkTab",
+    cats[0].items
+  );
+  const [searchText, setSearchText] = useLocalState(context, "seachText", "");
+  // Search equipment
+  const SelectEquipment = (cat, searchText = "") => {
+    const EquipmentSearch = createSearch(searchText, item => item.name);
+    return flow([
+      filter(item => item?.name),
+      searchText && filter(EquipmentSearch),
+      sortBy(item => item.name),
```

Comment:
Если в 102 строке проверяешь наличие `item`, то и тут, наверное, надо?
```suggestion
      sortBy(item => item?.name),
```

Reply:
Честно, это я скопипастил с блока ниже. Там оказывается была готовая реализация поиска. Можно и добавить